draft-ietf-httpbis-semantics-latest.xml

```diff
@@ -2628,49 +2634,103 @@ Host: www.example.org
       <td><xref target="header.accept-ranges"/></td>
     </tr>
   </tbody>
 </table>
 
-<section title="Byte Ranges" anchor="byte.ranges">
-  <x:anchor-alias value="bytes-unit"/>
+<section title="Range Specifiers" anchor="range.specifiers">
 <t>
-   Since representation data is transferred in payloads as a sequence of
-   octets, a byte range is a meaningful substructure for any representation
-   transferable over HTTP (<xref target="representations"/>). The "bytes" range unit is
-   defined for expressing subranges of the data's octet sequence.
+   Ranges are expressed in terms of a range unit paired with a set of range
+   specifiers. The range unit name determines what kinds of range-spec
+   are applicable to its own specifiers. Hence, the following gramar is
+   generic: each range unit is expected to specify requirements on when
+   <x:ref>int-range</x:ref>, <x:ref>suffix-range</x:ref>, and
+   <x:ref>other-range</x:ref> are allowed.
 </t>
-<sourcecode type="abnf7230"><iref primary="true" item="Grammar" subitem="bytes-unit"/>
-  <x:ref>bytes-unit</x:ref>       = "bytes"
-</sourcecode>
 <t anchor="rule.ranges-specifier">
-  <x:anchor-alias value="byte-range-set"/>
-  <x:anchor-alias value="byte-range-spec"/>
-  <x:anchor-alias value="byte-ranges-specifier"/>
-  <x:anchor-alias value="first-byte-pos"/>
-  <x:anchor-alias value="last-byte-pos"/>
   <x:anchor-alias value="ranges-specifier"/>
-   A byte-range request can specify a single range of bytes or a set
+  <x:anchor-alias value="range-set"/>
+  <x:anchor-alias value="range-spec"/>
+   A range request can specify a single range or a set
    of ranges within a single representation.
 </t>
-<sourcecode type="abnf7230"><iref primary="true" item="Grammar" subitem="ranges-specifier"/><iref primary="true" item="Grammar" subitem="byte-ranges-specifier"/><iref primary="true" item="Grammar" subitem="byte-range-set"/><iref primary="true" item="Grammar" subitem="byte-range-spec"/><iref primary="true" item="Grammar" subitem="first-byte-pos"/><iref primary="true" item="Grammar" subitem="last-byte-pos"/>
-  <x:ref>byte-ranges-specifier</x:ref> = <x:ref>bytes-unit</x:ref> "=" <x:ref>byte-range-set</x:ref>
-  <x:ref>byte-range-set</x:ref>  = 1#( <x:ref>byte-range-spec</x:ref> / <x:ref>suffix-byte-range-spec</x:ref> )
-  <x:ref>byte-range-spec</x:ref> = <x:ref>first-byte-pos</x:ref> "-" [ <x:ref>last-byte-pos</x:ref> ]
-  <x:ref>first-byte-pos</x:ref>  = 1*<x:ref>DIGIT</x:ref>
-  <x:ref>last-byte-pos</x:ref>   = 1*<x:ref>DIGIT</x:ref>
+<sourcecode type="abnf7230"><iref primary="true" item="Grammar" subitem="ranges-specifier"/><iref primary="true" item="Grammar" subitem="range-set"/><iref primary="true" item="Grammar" subitem="range-spec"/>
+  <x:ref>ranges-specifier</x:ref> = <x:ref>range-unit</x:ref> "=" <x:ref>range-set</x:ref>
+  <x:ref>range-set</x:ref>        = <x:ref>range-spec</x:ref> *( "," <x:ref>range-spec</x:ref>)
```

Comment:
Perhaps it is better to leave this as

   <x:ref>range-set</x:ref>        = 1#<x:ref>range-spec</x:ref>

until we have decided #212